### ISSUE TITLE
4.8.0 - opens any default browser with a separate icon

### ISSUE THREAD

**Issue by IvoAndonov** [linuxmint/cinnamon#9728]
* Cinnamon version: 4.8.0
   - Please specify if you are using the daily builds PPA: IDK - whatever comes from Arch repository, that's it.
 * Distribution - (Arch)
 * Graphics hardware *and* driver used: GTX 1660 Ti, driver version 450.57
 * Architecture: 64-bit

[xsession-errors.txt](https://github.com/linuxmint/cinnamon/files/5609960/xsession-errors.txt)
 
**Issue**
I've set my default browser (Waterfox) to startup along with the system. When session restarted (Ctrl+Alt+Backspace) or after system reboot, the browser is always opened as a separate process with a separate icon on the panel. As you can see from the screenshot named "bug.png", it opens the browser with its default icon and for unknown reason - as a separate process.
![bug](https://user-images.githubusercontent.com/41172201/100492380-1678ad80-3134-11eb-8416-cd439c8b02da.png)
The round blue icon with a fox on it is the icon I use for this browser. Normally, when the browser starts with the system, my custom icon should be marked as active/open:
![normal](https://user-images.githubusercontent.com/41172201/100492403-41630180-3134-11eb-84aa-c42c6dd2e99f.png)

Note 1: until a moment ago I had Cinnamon 4.6.4 and in that version such a thing never happened!
Note 2: this happens ONLY with the browser! I have other programs set to autolaunch, they have custom icons as well but they open just fine.

**Steps to reproduce**
1. Set whatever browser (Firefox, Waterfox, Chrome) as default browser.
2. Set a custom icon for it and pin the browser to the panel.
3. Set the browser to autostart with the system.
4. Restart the session or reboot the system and see the results: the browser will open as a separate window/process with its default icon.

**Expected behaviour**
The browser should open with the custom icon on the panel the user has set, like on the normal.png screenshot.

**Other information**
I can make a video with my phone, if the above information and screenshots are not enough to understand the problem.

**Comment by mtwebster** [comment]
This should be fixed already by https://github.com/linuxmint/cinnamon/commit/8171c084c572fdc0cd695eb90182101af0818b28 and will be in the next point release.

**Comment by IvoAndonov** [comment]
This bug returned in cinnamon 5.0.2 but is somewhat different than before bc now I found a workaround and it happens only after the browser has been updated. For instance, when I had Firefox 89.0 and it updated to 89.0.1, when I clicked on the pinned icon on the panel, the new version opened as a separate program. I thought it was an accident, so I didn't pay much attention to it. But an hour ago 89.0.2 and the same thing happened again - clicking on the icon for 89.0.1 opened 89.0.2 as a separate program.
I found a workaround which can be used until the problem is fixed:
1. When the browser updates, run it, so that it opens as a new prorgram.
2. Unpin from the panel the old icon.
3. Right click on the newly opened icon of the same browser and pin it to the panel. It will automatically take the place of the old icon. From now own the latest version will open as it should - as the same program, until the next version update.
